Document the limitations of the pip install --prefix argument (#11775)
pypa/pip: news/11775.doc.rst

> Cross-reference the --python flag in the docs for the --prefix flag, and mention the --prefix limitations with regards to installed console scripts.

Reviewer: ```suggestion
Cross-reference the ``--python`` flag from the ``--prefix`` flag,
and mention limitations of ``--prefix`` regarding script installation.
```
